libbeat/monitoring/report/report.go

```diff
@@ -25,6 +25,11 @@ import (
 	"github.com/elastic/beats/libbeat/common"
 )
 
+const (
+	ReportingFormatProduction = "production"
```

Comment:
I stumbled over this `production` name several times during the review. I understand that the naming comes from shipping to the "Production" cluster. At the same time a monitoring cluster is also a production cluster. Perhaps we can name this internally `bulk` and `xpack` as the reporting type to make the difference more obvious?

Reply:
Ahhhh naming 😄 

> Perhaps we can name this internally `bulk` and `xpack` as the reporting type to make the difference more obvious?

I went with the `production` vs. `monitoring` cluster terminology as it's the same terminology we use in stack monitoring discussions so it's a vocabulary that's more "obvious" to most of us on the team (even if it might be flawed). Seeing these terms in debug messages and such when troubleshooting issues would instantly make sense to us IMO.

> At the same time a monitoring cluster is also a production cluster. 

Indeed, this is a fair point. Perhaps `monitoring` vs. `data` cluster then? Again the same problem arises: that the monitoring cluster also stores data, after all 😄. Also, this terminology doesn't align with the terminology being used in stack monitoring discussions but it does remove the use of `production` (as the monitoring cluster is also a production cluster).

Naming is hard and so I'm open to changing the naming here if reviewers feel particularly strongly in any one direction.

Reply:
> Perhaps we can name this internally `bulk` and `xpack` as the reporting type to make the difference more obvious?

BTW, the xpack endpoint also has `bulk` in it, so seeing `bulk` in the code or in a debug message could be a bit ambiguous too and might require some digging.

How about we be really explicit about the names of these constants (and not care so much about their values) then? Something like:

```go
type ReportingFormat int

const (
  ReportingFormatXPackMonitoringBulk ReportingFormat = iota
  ReportingFormatBulk
)
```

Reply:
No strong feelings in any direction, just thought it's worth mentioning. I like your `iota` proposal.